src/address.rs

```diff
@@ -1,116 +1,90 @@
 lazy_static! {
     pub static ref DEFAULT_IPC_ACTOR_NAMESPACE_ID: ActorID = 1000u64;
 }
 
 #[derive(Clone, PartialEq, Eq, Debug, Hash, Serialize, Deserialize)]
 pub struct IPCAddress {
-    inner: Address,
-}
-
-impl fmt::Display for IPCAddress {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        self.inner.fmt(f)
-    }
+    subnet_id: SubnetID,
+    raw_address: Address,
 }
 
 impl IPCAddress {
     /// Generates new address using ID protocol
-    pub const fn new_id(id: u64) -> Self {
+    pub fn new_id(id: u64) -> Self {
+        let d = SubnetID::default();
         Self {
-            inner: Address::new_id(id),
+            subnet_id: d,
+            raw_address: Address::new_id(id),
         }
     }
 
     /// Generates new IPC address
     pub fn new(sn: &SubnetID, addr: &Address) -> Result<Self, Error> {
-        let sn = sn.to_bytes();
-        let addr = addr.to_bytes();
-        let sn_size_vec = to_leb_bytes(sn.len() as u64)?;
-        let sn_size: &[u8] = sn_size_vec.as_ref();
-        let addr_size_vec = to_leb_bytes(addr.len() as u64)?;
-        let addr_size: &[u8] = addr_size_vec.as_ref();
-        let sp = [sn_size, addr_size, sn.as_slice(), addr.as_slice()].concat();
-        // include in fixed-length container
-        let mut key = [0u8; MAX_ADDRESS_LEN];
-        key[..sp.len()].copy_from_slice(sp.as_slice());
         Ok(Self {
-            inner: Address::new_delegated(*DEFAULT_IPC_ACTOR_NAMESPACE_ID, &key)?,
+            subnet_id: sn.clone(),
+            raw_address: *addr,
         })
     }
 
     /// Generates new IPC address
     pub fn new_from_ipc(sn: &SubnetID, addr: &Self) -> Result<Self, Error> {
-        let sn = sn.to_bytes();
-        let addr = addr.to_bytes();
-        let sn_size_vec = to_leb_bytes(sn.len() as u64)?;
-        let sn_size: &[u8] = sn_size_vec.as_ref();
-        let addr_size_vec = to_leb_bytes(addr.len() as u64)?;
-        let addr_size: &[u8] = addr_size_vec.as_ref();
-        let sp = [sn_size, addr_size, sn.as_slice(), addr.as_slice()].concat();
-        // include in fixed-length container
-        let mut key = [0u8; MAX_ADDRESS_LEN];
-        key[..sp.len()].copy_from_slice(sp.as_slice());
         Ok(Self {
-            inner: Address::new_delegated(*DEFAULT_IPC_ACTOR_NAMESPACE_ID, &key)?,
+            subnet_id: sn.clone(),
+            raw_address: addr.raw_address,
         })
     }
 
     /// Returns subnets of a IPC address
     pub fn subnet(&self) -> Result<SubnetID, Error> {
-        let bz = self.inner.payload().to_raw_bytes();
-        let sn_size = from_leb_bytes(&[bz[0]]).unwrap() as usize;
-        let s =
-            String::from_utf8(bz[2..sn_size + 2].to_vec()).map_err(|_| Error::InvalidIPCAddr)?;
-        SubnetID::from_str(&s).map_err(|_| Error::InvalidIPCAddr)
+        Ok(self.subnet_id.clone())
     }
 
     /// Returns the raw address of a IPC address (without subnet context)
     pub fn raw_addr(&self) -> Result<Address, Error> {
```

Comment:
Should we rename it to `local_addr`? "Raw" may suggest it's a representation of the whole address...

Reply:
I am fine with this, but let's first agree on the API we want because this change will have to propagated all over the place.